nemo_curator/utils/split_large_file.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+
+def _split_pyarrow_table(table: pa.Table, target_size_mb: int) -> list[pa.Table]:
+    target_size_bytes = target_size_mb * 1024 * 1024
+    # Slice full table into two chunks
+    tables = [table.slice(0, table.num_rows // 2), table.slice(table.num_rows // 2, table.num_rows)]
+    results = []
+    for t in tables:
+        if t.nbytes > target_size_bytes:
+            # If above the target size, continue chunking until chunks
+            # are below the target size
+            results.extend(_split_pyarrow_table(t, target_size_mb=target_size_mb))
+        else:
+            results.append(t)
+    return results
+
+
+def _split_parquet_file_by_size(infile: str, outdir: str, target_size_mb: int, verbose: bool) -> None:
+    root, ext = os.path.splitext(infile)
+    if not ext:
+        ext = ".parquet"
+    outfile_prefix = os.path.basename(root)
+    table = pq.read_table(infile)
```

Comment:
nit, i think we can be slightly better about this by reading row group wise so that we are not constrained by memory, and then split at row group or combine row groups (or parts of it)

Reply:
```
current_chunk = None
current_chunk_size_remaining = target_size 
row_groups = pq.read_metadata(file).num_row_groups()
for row_group_content in read_row_group(file, row_group):
    if size(row_group_content) > current_chunk_size_remaining:
         desired_row_group, remaining_row_group = split_row_group(current_chunk_size_remaining)
         current_chunk.append(desired_row_group)

```